lib/components/table.vue

```diff
@@ -524,14 +533,25 @@
 
                 if (data.then && typeof data.then === 'function') {
                     // Provider returned Promise
-                    data.then((items) => {
+                    data.then(items => {
                         this._providerSetLocal(items);
                     });
                 } else {
                     // Provider returned Array data
                     this._providerSetLocal(data);
                 }
+            },
+            hasFormatter(field) {
+                return field.formatter && ((typeof (field.formatter) === 'function') || (typeof (field.formatter) === 'string'));
+            },
+            callFormatter(item, key, field) {
+                if (field.formatter && (typeof (field.formatter) === 'function'))
+                    return field.formatter(item[key]);
+
+                if (field.formatter && (typeof (this.$parent[field.formatter]) === 'function'))
+                    return this.$parent[field.formatter](item[key]);
```

Comment:
But making the extra args available would make rendering for "virtual" columns (columns that don't have a value in items) be possible (i.e. a virtual column that is comprised of data from different fields within the item record)

Maybe make the formatter function use this signature:

`formatter(value, item, key)`

And the user can then ignore the extra `item`, `key` args if not needed.